core/src/main/java/feast/core/model/FeatureTable.java

```diff
@@ -263,6 +266,84 @@ public void updateFromProto(
         .collect(Collectors.toSet());
   }
 
+  /**
+   * Return a boolean to indicate if FeatureTable contains all specified entities.
+   *
+   * @param entitiesFilter contain entities that should be attached to the FeatureTable
+   * @return boolean True if FeatureTable contains all entities in the entitiesFilter
+   */
+  public boolean hasAllEntities(List<String> entitiesFilter) {
+    List<String> allEntitiesName =
+        this.getEntities().stream().map(entity -> entity.getName()).collect(Collectors.toList());
+    return allEntitiesName.equals(entitiesFilter);
+  }
+
+  /**
+   * Returns a map of Feature references and Features if FeatureTable's Feature contains all labels
+   * in the labelsFilter
+   *
+   * @param labelsFilter contain labels that should be attached to FeatureTable's features
+   * @return Map of Feature references and Features
+   */
+  public Map<String, FeatureV2> getFeaturesByRef(Map<String, String> labelsFilter) {
+    Map<String, FeatureV2> validFeaturesMap;
+    List<FeatureV2> validFeatures;
+    if (labelsFilter.size() > 0) {
+      validFeatures = filterFeaturesByAllLabels(this.getFeatures(), labelsFilter);
+      validFeaturesMap = getFeaturesRefToFeaturesMap(validFeatures);
+      return validFeaturesMap;
+    }
+    validFeaturesMap = getFeaturesRefToFeaturesMap(List.copyOf(this.getFeatures()));
+    return validFeaturesMap;
+  }
+
+  /**
+   * Returns map for accessing features using their respective feature reference.
+   *
+   * @param features List of features to insert to map.
+   * @return Map of featureRef:feature.
+   */
+  public Map<String, FeatureV2> getFeaturesRefToFeaturesMap(List<FeatureV2> features) {
```

Comment:
why this method is public? Seems that it's being only used by `getFeaturesByRef`